test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -190,18 +176,18 @@
             pool-context (:pool-context context)]
         ;; set a ruby constant in each instance so that we can recognize them
         (is (true? (set-constants-and-verify pool-context 4)))
-        (let [flush-complete (add-watch-for-flush-complete pool-context)
-              ;; borrow an instance and hold the reference to it.
-              instance (jruby-protocol/borrow-instance jruby-service
-                                                       :hold-instance-while-pool-flush-in-progress-test)]
+        ;; borrow an instance and hold the reference to it.
+        (let [instance (jruby-protocol/borrow-instance
+                        jruby-service
+                        :hold-instance-while-pool-flush-in-progress-test)]
           ;; trigger a flush
           (jruby-protocol/flush-jruby-pool! jruby-service)
           ;; wait for the new pool to become available
           (is (true? (wait-for-new-pool jruby-service)))
           ;; return the instance
           (jruby-protocol/return-instance jruby-service instance :hold-instance-while-pool-flush-in-progress-test)
           ;; wait until the flush is complete
-          (is (true? (timed-deref flush-complete))
+          (is (true? (timed-await (:pool-agent pool-context)))
```

Comment:
Isn't there a different race here now?  If the flush completes before we hit the `timed-await`, then won't that cause a problem?